### PR TITLE
[ADD] udes_stock: Write destination location before qty done when picking the items

### PR DESCRIPTION
**Pull request by armandcela**
SE-527
Signed-off-by: Armand Cela <armand.cela@unipart.io>

### CODE REVIEW

File: addons/udes_stock/models/stock_move_line.py [hunk 1]
```diff
@@ -652,6 +651,8 @@ def _mark_as_done(self, values, split=True):
             )
 
         if split:
+            if values.get("location_dest_id"):
+                self.location_dest_id = values["location_dest_id"]
```
Comment (rajeeshknambiar) [inline addons/udes_stock/models/stock_move_line.py:655]: Isn't this trying to set `location_dest_id` object with integer id value?

**Review by rajeeshknambiar — APPROVED** [review]
LGTM

File: addons/udes_stock/models/stock_move_line.py (hunk 1 above)
Comment (armandcela) [reply addons/udes_stock/models/stock_move_line.py:655]: Yes when you write a many2one field it handles both integer or recordset

File: addons/udes_stock/models/stock_move_line.py [hunk 2]
```diff
@@ -262,13 +262,12 @@ def mark_as_done(
                     )
                 else:
                     ml_values["qty_done"] = ml.product_qty
+                if loc_dest_instance is not None:
+                    # HERE(ale): updating the dest loc here to have a single
+                    # invocation of its constraint handler (see below)
```
Comment (adamrpatrick) [inline addons/udes_stock/models/stock_move_line.py:267]: Comment no longer needed

**Review by adamrpatrick — APPROVED** [review]

File: addons/udes_stock/models/stock_move_line.py (hunk 2 above)
Comment (armandcela) [reply addons/udes_stock/models/stock_move_line.py]: Pushed the suggested change

**Pull request merged**